docs/cpan-dependency-confusion.md

```diff
@@ -0,0 +1,239 @@
+
+Since JFrog Artifactory does not support CPAN as a first class citizen but only a "proxy" CPAN (as in "HTTP proxy"), CPAN is not vulnerable via this artifact repository manager.
+
+### StratoPAN
+Not affected, not doing fallback on the public CPAN.
+
+## Mitigations
+Beyond technical CPAN client/options recommendations exposed earlier, sharing some more "shift left" guards and explicit security advice.
+
+### PAUSE operational model 
+The way CPAN and PAUSE work, and the very "social" nature of this ecosystem makes it very difficult to attack CPAN via dependency confusions. 
+
+1. New authors don't immediately get an account but are approved by an admin. With a priori approval like this, automatic creation of fake accounts is less likely.
+2. New uploads do not follow a priori control but are publicly reviewed. Malware code can be spotted and claiming namespaces with bad intention would likely be reported rapidly. 
+
+In this context, it seems difficult for an attacker/malware module to go unnoticed and exploit dependency confusion.
+
+### Isolate private requirements
+- Use different files for private and public requirements (`cpanfile.private` / `cpanfile`)
+- Do not specify private requirements in Makefile.PL, Build.PL, META.* or cpanfile/cpmfile
+
+Both previous solutions are sub-optimal
+
+### Naming
+#### Scoping like NPM
+Remediation for NPM ecosystem mentions [scoping](https://docs.npmjs.com/threats-and-mitigations#by-typosquatting--dependency-confusion) (naming packages like `@my-company/my-module`).
+A scope guarantees [protection of all names under it](https://docs.npmjs.com/package-scope-access-level-and-visibility) but also [can be tied to a registry](https://docs.npmjs.com/cli/v7/using-npm/scope#associating-a-scope-with-a-registry).
+ 
+The CPAN operating model works differently and has no such scopes. Technically, being the first person to publish a module offers FIRSTCOME permissions on the module name. Though it does not recusively protect all names under it, it effectively protects the fully qualified name forever. 
+
+Doing this ("placeholding") for private modules is an option, but it would be very unwise and very egoist to do that, effectively polluting the CPAN namespace for personal needs.
+Beyond these considerations, you need only publish once (a very small version that never will interfer with your local versioning) to claim the namespace. Remember, it's not a scope. Claiming `Module` does not protect `Module::Foo` nor `Module::Bar`.
+
+The module [install](https://metacpan.org/pod/install) somewhat fills this use case, like gem [bundle](https://rubygems.org/gems/bundle) in rubygems.
+
+#### Reserved namespaces
+Use reserved namespaces that can't resolve to public things:
+- `_Underscore::Trick` or `UN_DESCORE::Trick`
+- `Local::*` but as of today only reserved "by convention", see [Do not index Local::* (nor Local)](https://github.com/andk/pause/pull/541)
```

Comment:
"as of November 2024"

Reply:
I did think of adding that in my commit - probably should have

Reply:
Fixed